services/apps/src/apps/clients/iplant_groups.clj

```diff
@@ -1,14 +1,110 @@
 (ns apps.clients.iplant-groups
-  (:require [cemerick.url :as curl]
+  (:use [medley.core :only [remove-vals]]
+        [slingshot.slingshot :only [throw+]])
+  (:require [apps.util.config :as config]
+            [cemerick.url :as curl]
             [clj-http.client :as http]
-            [apps.util.config :as config]))
+            [clojure.string :as string]
+            [clojure-commons.exception :as cx]
+            [kameleon.uuids :refer [uuidify]]))
 
-(defn- lookup-subject-url
-  [short-username]
-  (str (curl/url (config/ipg-base) "subjects" short-username)))
+(def grouper-user "de_grouper")
+(def grouper-user-group-fmt "iplant:de:%s:users:de-users")
+(def grouper-app-permission-def-fmt "iplant:de:%s:apps:app-permission-def")
+(def grouper-app-resource-name-fmt "iplant:de:%s:apps:%s")
+
+(defn- grouper-user-group
+  []
+  (format grouper-user-group-fmt (config/env-name)))
+
+(defn- grouper-app-permission-def
+  []
+  (format grouper-app-permission-def-fmt (config/env-name)))
+
+(defn- grouper-app-resource-name
+  [app-id]
+  (format grouper-app-resource-name-fmt (config/env-name) app-id))
```

Comment:
I considered memoizing these, but it's cheap enough simply to format the strings every time.